singlePost.php

```diff
@@ -1,31 +1,66 @@
 <?php
+declare(strict_types=1);
+session_start();
+require_once 'src/services/PostDisplayService.php';
+require_once 'src/services/DatabaseConnectionService.php';
+require_once 'src/entities/PostEntity.php';
+require_once 'src/models/PostsModel.php';
+require_once 'src/entities/CommentEntity.php';
+require_once 'src/services/CommentValidationService.php';
 
- declare(strict_types=1);
- require_once 'src/services/PostDisplayService.php';
- require_once 'src/services/DatabaseConnectionService.php';
- require_once 'src/entities/PostEntity.php';
- require_once 'src/models/PostsModel.php';
+include_once 'header.php';
 
- $db = DatabaseConnectionService::connect();
- $PostsModel = new PostsModel($db);
+if (!isset($_SESSION['loggedIn'])) {
+    header('Location: login.php');
+    exit();
+}
+
+$db = DatabaseConnectionService::connect();
+$PostsModel = new PostsModel($db);
 
 $post = $PostsModel->getSingle((int)$_GET['id']);
 
-if (!$post){
+if (!$post) {
     header('Location: index.php');
 }
 
- require_once 'header.php';
-?>
+unset($_SESSION['contentError']);
+$content = $_POST['content'] ?? '';
+
+if ($_SERVER["REQUEST_METHOD"] === "POST" && isset($_SESSION['username_id'])) {
+    $contentValid = CommentValidationService::ContentValidation($content);
+    if (!$contentValid) {
+        $_SESSION['contentError'] = 'Content must be between 10 and 200 characters.';
+    }
 
+    if ($contentValid) {
+        $PostModel = new PostsModel($db);
+        CommentValidationService::AddCommentToDatabase($content, $PostModel,(int)$_GET['id'] );
```

Comment:
Services don't add stuff to the database. Which kind of class is used for database operations?